### DISCUSSION TITLE
Any plans on supporting suspense combined with keepPreviousData?

### DISCUSSION THREAD

**Discussion by WagnerMoreira** [vercel/swr#2587] [General] [answered]
> I'm aware that suspense currently does not work with keepPreviousData or a middleware like the `laggy` from the docs example because `swr.data` is never undefined while using suspense so the features that rely on this state do not work.

I'm doing a migration on a project that had swr 0.4.x to the latest 2.1.5, and I need to use the keepPreviousData option a lot, I ended up using middleware to achieve the same because of the granular control. The project uses suspense everywhere, and there is a `<Suspense>` at the root level, route levels, etc... so I'm being forced to disable suspense in some places because I need to use keepPreviousData, but that causes weird behaviors since suspense is disabled, React Query supports this so it is possible.

Would be more than wonderful if SWR could support suspense and keepPreviousData/laggy middleware at the same time.
if anyone is curious to see the differences, I created a repo to illustrate this https://github.com/WagnerMoreira/react-query-keep-prev-data

**Comment by WagnerMoreira** [comment]
bumping this for visibility!

**Reply by promer94** [reply, marked as answer]
https://github.com/vercel/swr/pull/2649
